Core/PrimaryGenerator.cpp

```diff
@@ -182,7 +234,9 @@ void PrimaryGenerator::GenerateEvtSmallChamber(G4Event* event, const G4double ef
 
   if (evtFractions[0] > random) // pPs
   {
-    event->AddPrimaryVertex(GenerateTwoGammaVertex( vtxPosition, T0, 
+    //event->AddPrimaryVertex(GenerateTwoGammaVertex( vtxPosition, T0, 
+    //                        material->GetLifetime(random, MaterialExtension::DecayChannel::Para2G)));
+    event->AddPrimaryVertex(GenerateThreeGammaVertexPara( vtxPosition, T0, 
```

Comment:
As I look at this, maybe it would be nice to introduce additional DecayChannel variable in MaterialExtension class, and additional if condition.
Could You add in MaterialExtension.h in line 36 add after Para2G: 
Para3G,
Then in MaterialExtension.cpp add after line 71:
else if( channel == DecayChannel::Para3G )
    return fMaterialParameters->GetpPsLifetime();
then You can change a little functions
GenerateEvtSmallChamber and GenerateEvtLargeChamber
with additional if(evtFraction[0] + evt... > random)
so there is separate condition for pPs into 3.
But maybe some more details You can ask via mail.